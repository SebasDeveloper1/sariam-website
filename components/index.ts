import Button from './Buttons/Button/Button';
import IconButton from './Buttons/IconButton/IconButton';
import SliderComponent from './SliderComponent/SliderComponent';
import ServiceCard from './ServiceCard/ServiceCard';
export { Button, IconButton, SliderComponent, ServiceCard };
