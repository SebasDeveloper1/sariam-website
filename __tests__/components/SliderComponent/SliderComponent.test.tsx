import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { SliderComponent } from '@/components';

const eventsList: string[] = [
  'Eventos empresariales',
  '15 Años',
  'Matrimonios',
  'Bautizos',
  'Cumpleaños',
  'Fotografía',
  'Luces y Sonido',
  'Animación y Dj',
  'Eventos empresariales',
  '15 Años',
  'Matrimonios',
  'Bautizos',
  'Cumpleaños',
  'Fotografía',
  'Luces y Sonido',
  'Animación y Dj',
];

describe('SliderComponent component', () => {
  describe('Rendering', () => {
    // Test that the component renders without errors
    test('Should render component', () => {
      const { getByRole } = render(<SliderComponent />);
      const slider = getByRole('slider');
      expect(slider).toBeInTheDocument();
    });

    // Test that the component displays three carousels with events, each event in a rounded rectangle with a lime green and gray color scheme
    test('Should show carousel container', () => {
      const { container } = render(<SliderComponent />);
      const carousel = container.querySelector('.animate-carousel');
      expect(carousel).toBeInTheDocument();
    });

    test('Should show carousel items', () => {
      const { container } = render(<SliderComponent />);
      const carouselItems = container.querySelectorAll('.span_lg');
      expect(carouselItems.length).toBe(eventsList.length * 3);
      carouselItems.forEach((carouselItem) => {
        expect(carouselItem).toHaveClass('bg-lime-900/50');
        expect(carouselItem).toHaveClass('text-gray-50');
        expect(carouselItem).toHaveClass('rounded-2xl');
      });
    });
  });

  describe('Snapshots', () => {
    // Test that the rendered SliderComponent matches the snapshot
    test('should match the SliderComponent component snapshot', () => {
      const { getByRole } = render(<SliderComponent />);
      const slider = getByRole('slider');
      expect(slider).toMatchSnapshot();
    });
  });
});
