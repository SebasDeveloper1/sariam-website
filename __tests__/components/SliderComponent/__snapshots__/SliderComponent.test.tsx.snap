// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SliderComponent component Snapshots should match the SliderComponent component snapshot 1`] = `
<div
  class="flex items-center justify-center w-full py-2 bg-lime-600"
  role="slider"
>
  <div
    class="w-[300%] h-12 overflow-hidden relative"
  >
    <div
      class="w-[300%] flex items-center h-12 justify-around absolute left-0 animate-carousel gap-4"
    >
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          Eventos empresariales
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          15 Años
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          Matrimonios
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          Bautizos
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          Cumpleaños
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          Fotografía
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          Luces y Sonido
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          Animación y Dj
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          Eventos empresariales
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          15 Años
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          Matrimonios
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          Bautizos
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          Cumpleaños
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          Fotografía
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          Luces y Sonido
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          Animación y Dj
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          Eventos empresariales
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          15 Años
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          Matrimonios
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          Bautizos
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          Cumpleaños
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          Fotografía
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          Luces y Sonido
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          Animación y Dj
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          Eventos empresariales
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          15 Años
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          Matrimonios
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          Bautizos
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          Cumpleaños
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          Fotografía
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          Luces y Sonido
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          Animación y Dj
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          Eventos empresariales
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          15 Años
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          Matrimonios
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          Bautizos
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          Cumpleaños
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          Fotografía
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          Luces y Sonido
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          Animación y Dj
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          Eventos empresariales
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          15 Años
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          Matrimonios
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          Bautizos
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          Cumpleaños
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          Fotografía
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          Luces y Sonido
        </span>
      </div>
      <div>
        <span
          class="span_lg block w-fit px-6 py-2 rounded-2xl text-gray-50 whitespace-nowrap bg-lime-900/50"
          style="margin-right: 1rem;"
        >
          Animación y Dj
        </span>
      </div>
    </div>
  </div>
</div>
`;
